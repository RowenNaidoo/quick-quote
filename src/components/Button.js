import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Button = ({ buttonText, onButtonClicked }) => (
  <StyledButton onClick={onButtonClicked}>
    {buttonText}
  </StyledButton>
)

const StyledButton = styled.button`
  border-color: #047db1;
  color: #fff;
  background-color: #047db1;
  border: 1px solid transparent;
  margin: 16px 0;
  max-width: 210px;
  min-height: 44px;
  padding: 10px 0 14px 0;
  text-align: center;
  border-radius: 50px;
  cursor: pointer;
  width: 100%;
`

Button.propTypes = {
  buttonText: PropTypes.string,
  onButtonClicked: PropTypes.func
}

export default Button