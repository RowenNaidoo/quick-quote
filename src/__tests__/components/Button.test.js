import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import Button from '../../components/Button'


describe('<Button />', () => {
  let component
  let mockButtonClicked = jest.fn()

  const defaultProps = {
    buttonText: 'click me',
    onButtonClicked: mockButtonClicked
  }

  beforeEach(() => {
    component = shallow(<Button {...defaultProps} />)
  })

  test('matches snapshot', () => {
    const component = renderer.create(<Button {...defaultProps} />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('shows button text', () => {
    expect(component.text()).toEqual('click me')
  })

  test('fires onClick function', () => {
    component.simulate('click')
    expect(mockButtonClicked).toHaveBeenCalled()
  })
})