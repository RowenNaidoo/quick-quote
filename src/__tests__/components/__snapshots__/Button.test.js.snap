// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<button
  className="sc-AxjAm bJArkW"
  onClick={[MockFunction]}
>
  click me
</button>
`;
